src/sidebar/util/oauth-client.js

```diff
@@ -213,11 +184,44 @@ export default class OAuthClient {
     const headers = {
       'Content-Type': 'application/x-www-form-urlencoded',
     };
-    return fetch(url, {
+    const response = await fetch(url, {
       method: 'POST',
       headers,
       body: params.toString(),
     });
+
+    if (response.status !== 200) {
+      throw new Error('Request failed');
```

Comment:
Maybe a bit more specific? (there are many other network errors, offline, etc.)

```suggestion
      throw new Error('Request failed with a non-200 status');
```

Reply:
I have added the status code to the error, although this is expected to be pretty short-lived code that is replaced in the next PR in the series.